tesseract_kinematics/core/src/kinematics_plugin_factory.cpp

```diff
@@ -244,6 +244,17 @@ KinematicsPluginFactory::createFwdKin(const std::string& solver_name,
                                       const tesseract_scene_graph::SceneGraph& scene_graph,
                                       const tesseract_scene_graph::SceneState& scene_state) const
 {
+#ifdef TESSERACT_PLUGIN_FACTORY_CALLBACKS
```

Comment:
I think it might be best to make this either or implementation.
``` c++
#ifndef TESSERACT_PLUGIN_FACTORY_CALLBACKS
DiscreteContactManager::UPtr
ContactManagersPluginFactory::createDiscreteContactManager(const std::string& name,
                                                           const tesseract_common::PluginInfo& plugin_info) const
{
  try
  {
    auto it = discrete_factories_.find(plugin_info.class_name);
    if (it != discrete_factories_.end())
      return it->second->create(name, plugin_info.config);

    auto plugin = plugin_loader_.instantiate<DiscreteContactManagerFactory>(plugin_info.class_name);
    if (plugin == nullptr)
    {
      CONSOLE_BRIDGE_logWarn("Failed to load symbol '%s'", plugin_info.class_name.c_str());
      return nullptr;
    }
    discrete_factories_[plugin_info.class_name] = plugin;
    return plugin->create(name, plugin_info.config);
  }
  catch (const std::exception&)
  {
    CONSOLE_BRIDGE_logWarn("Failed to load symbol '%s'", plugin_info.class_name.c_str());
    return nullptr;
  }
}
#else
DiscreteContactManager::UPtr
ContactManagersPluginFactory::createDiscreteContactManager(const std::string& name,
                                                           const tesseract_common::PluginInfo& plugin_info) const
{

  if (discrete_manager_create_callback_)
  {
    auto cm = discrete_manager_create_callback_(plugin_info.class_name);
    if (cm)
    {
      return cm->create(name, plugin_info.config);
    }
  }

    CONSOLE_BRIDGE_logWarn("Failed to create plugin '%s' using callback", plugin_info.class_name.c_str());
    return nullptr;
}
#endif
```